.gitlab-ci.yml

```diff
@@ -2,34 +2,10 @@ variables:
   # No need for TLS because everything runs on the same machine
   DOCKER_TLS_CERTDIR: ""
 
-.build: &build
-  image: docker
-  variables:
-    CONTAINER_NAME: "$CI_REGISTRY/satoshilabs/trezor/trezor-user-env/trezor-user-env"
-  services:
-    - docker:dind
-  before_script:
-    - docker login $CI_REGISTRY -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD
+stages:
+  - prebuild
+  - build
 
-environment:
-  <<: *build
-  only:
-    - schedules  # nightly build
-    - master
-    - /^run\//
-  script:
-    - docker build --tag $CONTAINER_NAME:$CI_COMMIT_SHA --tag $CONTAINER_NAME:latest -f docker/Dockerfile .
-    - docker push $CONTAINER_NAME:$CI_COMMIT_SHA
-    - docker push $CONTAINER_NAME:latest
-
-environment test:
-  <<: *build
-  except:
-    - schedules
-    - master
-    - /^run\//
-  script:
-    # note that this way "test" tag gets overwritten everytime
-    - docker build --tag $CONTAINER_NAME:$CI_COMMIT_SHA --tag $CONTAINER_NAME:test -f docker/Dockerfile .
-    - docker push $CONTAINER_NAME:$CI_COMMIT_SHA
-    - docker push $CONTAINER_NAME:test
+include:
+  - ci/prebuild.yml
+  - ci/build.yml
```

Comment:
Mostly an OCD thing and probably from @vdovhanych but please let's have a newline at the end in each file.